utf8parse/src/lib.rs

```diff
@@ -91,3 +91,43 @@ impl Parser {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io::prelude::*;
+    use std::fs::File;
+    use Receiver;
+    use Parser;
+
+    impl Receiver for String {
+        fn codepoint(&mut self, c: char) {
+            self.push(c);
+        }
+
+        fn invalid_sequence(&mut self) {
+        }
+    }
+
+    #[test]
+    fn utf8parse_test() {
+        let mut buffer = String::new();
+        let mut file = File::open("src/UTF-8-demo.txt").unwrap();
+        let mut parser = Parser::new();
+
+        // assigning to _ so it won't matter that I'm not using the returned Result
+        // just want to read the file to a buffer
+        let _ = file.read_to_string(&mut buffer);
```

Comment:
We should probably call `unwrap()` or `expect()` on this result in case there actually is a problem; it would otherwise be more difficult to figure out where the test went wrong.